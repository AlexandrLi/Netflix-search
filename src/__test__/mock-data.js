export const movies = [{
    vote_count: 8410,
    id: 122,
    video: false,
    vote_average: 8.2,
    title: 'The Lord of the Rings: The Return of the King',
    popularity: 22.480443,
    poster_path: '/uexxR7Kw1qYbZk0RYaF9Rx5ykbj.jpg',
    original_language: 'en',
    original_title: 'The Lord of the Rings: The Return of the King',
    genre_ids: [
      12,
      14,
      28
    ],
    backdrop_path: '/8BPZO0Bf8TeAy8znF43z8soK3ys.jpg',
    adult: false,
    overview: 'Aragorn is revealed as the heir to the ancient kings as he, Gandalf and the other members of the broken fellowship struggle to save Gondor from Sauron\'s forces. Meanwhile, Frodo and Sam bring the ring closer to the heart of Mordor, the dark lord\'s realm.',
    release_date: '2003-12-01'
  },
  {
    vote_count: 9090,
    id: 120,
    video: false,
    vote_average: 8.1,
    title: 'The Lord of the Rings: The Fellowship of the Ring',
    popularity: 36.635241,
    poster_path: '/bxVxZb5O9OxCO0oRUNdCnpy9NST.jpg',
    original_language: 'en',
    original_title: 'The Lord of the Rings: The Fellowship of the Ring',
    genre_ids: [
      12,
      14,
      28
    ],
    backdrop_path: '/pIUvQ9Ed35wlWhY2oU6OmwEsmzG.jpg',
    adult: false,
    overview: 'Young hobbit Frodo Baggins, after inheriting a mysterious ring from his uncle Bilbo, must leave his home in order to keep it from falling into the hands of its evil creator. Along the way, a fellowship is formed to protect the ringbearer and make sure that the ring arrives at its final destination: Mt. Doom, the only place where it can be destroyed.',
    release_date: '2001-12-18'
  },
  {
    vote_count: 7842,
    id: 121,
    video: false,
    vote_average: 8,
    title: 'The Lord of the Rings: The Two Towers',
    popularity: 24.591994,
    poster_path: '/wf3v0Pn09jnT5HSaYal7Ami3bdA.jpg',
    original_language: 'en',
    original_title: 'The Lord of the Rings: The Two Towers',
    genre_ids: [
      12,
      14,
      28
    ],
    backdrop_path: '/dG4BmM32XJmKiwopLDQmvXEhuHB.jpg',
    adult: false,
    overview: 'Frodo and Sam are trekking to Mordor to destroy the One Ring of Power while Gimli, Legolas and Aragorn search for the orc-captured Merry and Pippin. All along, nefarious wizard Saruman awaits the Fellowship members at the Orthanc Tower in Isengard.',
    release_date: '2002-12-18'
  }
]

export const yodaMovies = [{
    vote_count: 1,
    id: 371796,
    video: false,
    vote_average: 7,
    title: 'The New Yoda Chronicles: Escape from the Jedi Temple',
    popularity: 1.677475,
    poster_path: '/2nwar7SCeY9fktHRA8LFwHRBqiI.jpg',
    original_language: 'en',
    original_title: 'The New Yoda Chronicles: Escape from the Jedi Temple',
    genre_ids: [
      16,
      10751
    ],
    backdrop_path: null,
    adult: false,
    overview: '',
    release_date: '2014-12-09'
  },
  {
    vote_count: 1,
    id: 424492,
    video: false,
    vote_average: 7,
    title: 'The New Yoda Chronicles: Clash of the Skywalkers',
    popularity: 1.053418,
    poster_path: '/d0nFx7eQvXsGZgRkVLY7NymtZXd.jpg',
    original_language: 'en',
    original_title: 'The New Yoda Chronicles: Clash of the Skywalkers',
    genre_ids: [
      16
    ],
    backdrop_path: null,
    adult: false,
    overview: 'The enraged Emperor sends Darth Vader and his entire fleet to find and destroy Luke as revenge for his destruction of the Holocrons. Luke, however, finds inspiration from the last remaining Holocron -- a daring move pulled off years earlier by his father, Anakin. The emboldened Luke uses Anakin\'s daring move to lure Vader and the Imperial Fleet into a trap that could turn the tide for the Rebellion - or lead to disaster.',
    release_date: '2014-11-23'
  }
]

export const movie = {
  adult: false,
  backdrop_path: '/fUjPnCmmBVHLNdgz2sqb1zp39Pr.jpg',
  belongs_to_collection: {
    id: 1241,
    name: 'Harry Potter Collection',
    poster_path: '/eVPs2Y0LyvTLZn6AP5Z6O2rtiGB.jpg',
    backdrop_path: '/wfnMt6LGqYHcNyOfsuusw5lX3bL.jpg'
  },
  budget: 125000000,
  genres: [{
      id: 12,
      name: 'Adventure'
    },
    {
      id: 14,
      name: 'Fantasy'
    },
    {
      id: 10751,
      name: 'Family'
    }
  ],
  homepage: 'http://harrypotter.warnerbros.com/harrypotterandthedeathlyhallows/mainsite/index.html',
  id: 671,
  imdb_id: 'tt0241527',
  original_language: 'en',
  original_title: 'Harry Potter and the Philosopher\'s Stone',
  overview: 'Harry Potter has lived under the stairs at his aunt and uncle\'s house his whole life. But on his 11th birthday, he learns he\'s a powerful wizard -- with a place waiting for him at the Hogwarts School of Witchcraft and Wizardry. As he learns to harness his newfound powers with the help of the school\'s kindly headmaster, Harry uncovers the truth about his parents\' deaths -- and about the villain who\'s to blame.',
  popularity: 32.913022,
  poster_path: '/dCtFvscYcXQKTNvyyaQr2g2UacJ.jpg',
  production_companies: [{
      name: '1492 Pictures',
      id: 436
    },
    {
      name: 'Warner Bros.',
      id: 6194
    },
    {
      name: 'Heyday Films',
      id: 7364
    }
  ],
  production_countries: [{
      iso_3166_1: 'GB',
      name: 'United Kingdom'
    },
    {
      iso_3166_1: 'US',
      name: 'United States of America'
    }
  ],
  release_date: '2001-11-16',
  revenue: 976475550,
  runtime: 152,
  spoken_languages: [{
    iso_639_1: 'en',
    name: 'English'
  }],
  status: 'Released',
  tagline: 'Let the Magic Begin.',
  title: 'Harry Potter and the Philosopher\'s Stone',
  video: false,
  vote_average: 7.5,
  vote_count: 7441
};

export const anotherMovie = {
  vote_count: 9090,
  id: 120,
  video: false,
  vote_average: 8.1,
  title: 'The Lord of the Rings: The Fellowship of the Ring',
  popularity: 36.635241,
  poster_path: '/bxVxZb5O9OxCO0oRUNdCnpy9NST.jpg',
  original_language: 'en',
  original_title: 'The Lord of the Rings: The Fellowship of the Ring',
  genre_ids: [
    12,
    14,
    28
  ],
  backdrop_path: '/pIUvQ9Ed35wlWhY2oU6OmwEsmzG.jpg',
  adult: false,
  overview: 'Young hobbit Frodo Baggins, after inheriting a mysterious ring from his uncle Bilbo, must leave his home in order to keep it from falling into the hands of its evil creator. Along the way, a fellowship is formed to protect the ringbearer and make sure that the ring arrives at its final destination: Mt. Doom, the only place where it can be destroyed.',
  release_date: '2001-12-18'
}

export const FAKE_ACTION = 'FAKE_ACTION';

export const initialState = {
  movies: [],
  selectedMovie: null,
  sortBy: 'release'
};