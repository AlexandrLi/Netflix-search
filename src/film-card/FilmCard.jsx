import React from 'react';
import s from './film-card.css';

export class FilmCard extends React.Component {

  render() {
    return (
      <div className={s.card}>
        <img src={`https://image.tmdb.org/t/p/w342${this.props.movie.poster_path}`} alt="no image" />
        <div className={s.row}>
          <h2>{this.props.movie.title}</h2>
          <div>
            {!!this.props.movie.release_date && <div className={s.year}>{this.props.movie.release_date.slice(0, 4)}</div>}
          </div>
        </div>
        <p>{this.props.movie.vote_average}</p>
      </div>
    )
  }
}