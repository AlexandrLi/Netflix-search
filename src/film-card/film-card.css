.card {
  display: flex;
  flex-direction: column;
  width: 250px;
  padding: 20px 10px;
}

.card img {
  min-height: 320px;
  max-height: 320px;
}

.row {
  display: flex;
  justify-content: space-between;
  margin: 10px 0;
  font-size: 12px;
}

.row h2 {
  font-size: 16px;
  text-transform: uppercase;
  padding-right: 5px;
}

.year {
  border: 1px solid black;
  padding: 2px;
}

p {
  font-size: 12px;
}