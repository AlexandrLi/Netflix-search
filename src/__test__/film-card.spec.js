import React from "react"
import renderer from "react-test-renderer"
import { shallow } from "enzyme";
import { FilmCard } from "../film-card/FilmCard";
import { movie } from './mock-data';

describe("Film-card snapshot", () => {
  it("Take film-card snapshot", () => {
    const renderedFilmCard = renderer.create(<FilmCard movie={movie} />).toJSON();
    expect(renderedFilmCard).toMatchSnapshot();
  })
});

describe("Shalow render of Film-card component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<FilmCard movie={movie} />)
  });

  it("Component successfully rendered", () => {
    expect(component.length).toEqual(1);
  });

  it("Poster url", () => {
    expect(component.find("img").prop("src"))
      .toEqual(`https://image.tmdb.org/t/p/w342${movie.poster_path}`);
  });

  it("contains correct movie title", () => {
    expect(component.find('h2').get(0).props.children).toBe(movie.title);
  });

  it("contains movie release date", () => {
    expect(component.find('.year').get(0).props.children).toBe("2001");
  });

  it("contains movie votes", () => {
    expect(component.find('p').get(0).props.children).toBe(movie.vote_average);
  });
});