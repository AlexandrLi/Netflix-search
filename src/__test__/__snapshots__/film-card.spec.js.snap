// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film-card snapshot Take film-card snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="no image"
    src="https://image.tmdb.org/t/p/w342/dCtFvscYcXQKTNvyyaQr2g2UacJ.jpg"
  />
  <div
    className="row"
  >
    <h2>
      Harry Potter and the Philosopher's Stone
    </h2>
    <div>
      <div
        className="year"
      >
        2001
      </div>
    </div>
  </div>
  <p>
    7.5
  </p>
</div>
`;
